### PR TITLE
chore(authenticator): Clean up

### PR DESCRIPTION
**Pull request by dnys1**
*Issue #, if available:*

*Description of changes:*
- Refactor widget hierarchy
- Single view model
- Linting rules
- Update example
- Other things

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_authenticator/analysis_options.yaml [hunk 1]
```diff
@@ -1,14 +1,8 @@
-analyzer:
-  strong-mode:
-    implicit-dynamic: false
+include: package:amplify_lints/flutter_lib.yaml
```
Comment (haverchuck) [inline packages/amplify_authenticator/analysis_options.yaml:1]: +1

File: packages/amplify_authenticator/lib/amplify_authenticator.dart [hunk 2]
```diff
@@ -15,113 +15,79 @@
...
-  ///     - Email
-  ///     - Phone Number
-  /// 3. Confirm Sign in
-  ///     - define
-  ///     - define
-  ///     - define
+  /// {@macro authenticator.authenticator}
+  const Authenticator({
+    Key? key,
+    this.usernameAlias = Alias.username,
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/amplify_authenticator.dart:77]: Fyi, we will be doing away with `usernameAlias` shortly.

File: packages/amplify_authenticator/lib/amplify_authenticator.dart [hunk 3]
```diff
@@ -382,3 +278,66 @@ class _AuthenticatorState extends State<Authenticator> {
...
+          stream: stateMachineBloc.stream,
+          builder: (context, snapshot) {
+            final state = snapshot.data ?? const AuthLoading();
+            final Widget screen;
+            if (state is AuthLoading || state is AuthLoaded) {
+              screen = const LoadingScreen();
+            } else if (state is Authenticated) {
+              return Theme(data: userAppTheme, child: child);
+            } else if (state is AuthFlow) {
+              screen = AuthenticatorScreen(screen: state.screen);
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/amplify_authenticator.dart:313]: +1

File: packages/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart [hunk 4]
```diff
@@ -95,179 +104,152 @@ class StateMachineBloc {
...
         case 'CONFIRM_SIGN_IN_WITH_NEW_PASSWORD':
-          yield AuthFlow(
-            screen: AuthScreen.confirmSignInNewPassword,
-          );
-
+          yield AuthFlow.confirmSigninNewPassword;
           break;
         case 'RESET_PASSWORD':
-          yield AuthFlow(screen: AuthScreen.sendCode);
+          yield AuthFlow.sendCode;
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart:153]: +1

File: packages/amplify_authenticator/lib/src/utils/validators.dart [hunk 5]
```diff
@@ -69,43 +68,43 @@ Function validateSignUpPassword = (BuildContext context) {
...
       if (passwordCharacters
               .contains(PasswordPolicyCharacters.requiresSymbols) &&
           !_symbols.hasMatch(password) &&
           !password.contains("'")) {
-        passwordHints
-            .add('* ${_hintStrings.password_requires_symbols(context)}');
+        passwordHints.add('* ${_hintStrings.passwordRequiresSymbols(context)}');
       }
     }
     return null;
```
Comment (haverchuck) [inline packages/amplify_authenticator/lib/src/utils/validators.dart:92]: I think something got messed up when I merged this code... can you please replace `return null` in line 92 with 
`return passwordHints.isNotEmpty ? passwordHints.join('\n') : null`?

**Review by haverchuck — APPROVED** [review]

File: packages/amplify_authenticator/lib/src/utils/validators.dart (hunk 5 above)
Comment (dnys1) [reply packages/amplify_authenticator/lib/src/utils/validators.dart]: Fixed 👍

**Review by Jordan-Nelson — APPROVED** [review]

**Pull request merged**